src/storage/index/hash_index.cpp

```diff
@@ -659,11 +622,19 @@ PrimaryKeyIndex::PrimaryKeyIndex(const DBFileIDAndName& dbFileIDAndName, bool re
             hashIndexHeadersForReadTrx.end());
         KU_ASSERT(headerIdx == NUM_HASH_INDEXES);
     }
+    hashIndexDiskArrays = std::make_unique<DiskArrayCollection>(*fileHandle,
+        dbFileIDAndName.dbFileID, &bufferManager, wal,
+        INDEX_HEADER_PAGES /*firstHeaderPage follows the index header pages*/, true /*bypassWAL*/);
 
     if (keyDataTypeID == PhysicalTypeID::STRING) {
         overflowFile = std::make_unique<OverflowFile>(dbFileIDAndName, &bufferManager, wal,
             readOnly, vfs, context);
     }
+    if (newIndex) {
+        for (size_t i = 0; i < NUM_HASH_INDEXES * 2; i++) {
```

Comment:
Can u add a comment here on `NUM_HASH_INDEXES * 2`? one for each pSlot and oSlot disk array.